odoo/orm/fields.py

```diff
@@ -1614,7 +1631,7 @@ def _filter_not_equal(self, records: ModelType, cache_value: typing.Any) -> Mode
         """ Return the subset of ``records`` for which the value of ``self`` is
         either not in cache, or different from ``cache_value``.
         """
-        field_cache = self._get_cache(records.env)
+        field_cache = self._get_cache(records.env, to_write=True)
```

Comment:
My point is that this API doesn't make sense.  This call uses `to_write=True` to _read_ the cache.  From The flag `to_write` looks like a way to prevent an error than to document local intention.  And by the way, the standard implementation of `_has_field_access()` does not even take the operation into account.